File: src/main/java/Etc/Factorial.java
package Etc;

public class Factorial {

	public static void main(String[] args) {
		System.out.println(factorial(4));
	}

	public static int factorial(int input) {
		if (input <= 1) {
			return input;
		} else {
			return factorial(input - 1) * input;
		}

	}
}

File: 03_Template/recursion/Factorial.py
# 1 ~ n까지의 곱 구하기
def factorial(n:int):
    # 종료조건 : n이 0이라면 1을 반환한다.
    if n == 0:
        return 1
    # n이 0보다 크다면 현재의 n값에 재귀함수(n - 1)을 곱한다.
    else:
        return n * factorial(n - 1)
        
# 검증
print(factorial(4))